tests/unit/testsuite/Bolt/Boltpay/controllers/ShippingControllerTest.php

```diff
@@ -2,69 +2,318 @@
+        // Create some dummy product:
+        self::$_productIds = array
+        (
+            Bolt_Boltpay_ProductProvider::createDummyProduct('PHPUNIT_SC_TEST_PROD_01'),
+            Bolt_Boltpay_ProductProvider::createDummyProduct('PHPUNIT_SC_TEST_PROD_02'),
+            Bolt_Boltpay_ProductProvider::createDummyProduct('PHPUNIT_SC_TEST_PROD_03'),
+        );
+
     }
 
-    public function testPOBoxDoesNotMatchCases()
+    /**
+     * Delete dummy products after the test
+     */
+    public static function tearDownAfterClass()
     {
-        $doesAddressContainPOBoxMethod = new ReflectionMethod($this->_shippingController, 'doesAddressContainPOBox');
-        $doesAddressContainPOBoxMethod->setAccessible(true);
-        
-        $address1 = 'Post street';
-        $address2 = '2 Box';
-        $address3 = '425 Sesame St';
-        $check1 = $doesAddressContainPOBoxMethod->invoke($this->_shippingController, $address1);
-        $check2 = $doesAddressContainPOBoxMethod->invoke($this->_shippingController, $address2);
-        $check3 = $doesAddressContainPOBoxMethod->invoke($this->_shippingController, $address3);
-        $additionalCheck1 = $doesAddressContainPOBoxMethod->invoke($this->_shippingController, $address1, $address2);
-        $additionalCheck2 = $doesAddressContainPOBoxMethod->invoke($this->_shippingController, $address2, $address3);
-
-        $this->assertNotTrue($check1);
-        $this->assertNotTrue($check2);
-        $this->assertNotTrue($check3);
-        $this->assertNotTrue($additionalCheck1);
-        $this->assertNotTrue($additionalCheck2);
+        foreach (self::$_productIds as $productId) {
+            Bolt_Boltpay_ProductProvider::deleteDummyProduct($productId);
+        }
+
+        Mage::app()->getCache()->clean('matchingAnyTag', array('BOLT_QUOTE_PREFETCH'));
+    }
+
+    /**
+     * Test to see if cache is in a valid state after prefetch data is sent.  Prior to
+     * the prefetch, there should be no cache data.  After the prefetch, there should be
+     * cached data.
+     *
+     * @throws ReflectionException      on unexpected problems with reflection
+     * @throws Zend_Cache_Exception     on unexpected problems reading or writing to Magento cache
+     */
+    public function testIfEstimateIsCachedAfterPrefetch() {
+
+        Mage::app()->getCache()->clean('matchingAnyTag', array('BOLT_QUOTE_PREFETCH'));
+
+        /** @var Mage_Sales_Model_Quote $quote */
+        $quote = Mage::getSingleton('checkout/session')->getQuote();
+
+        $quote
+            ->removeAllAddresses()
+            ->removeAllItems()
+            ->setCustomerId(25)
+            ->setCustomerTaxClassId(3);
+
+        foreach(self::$_productIds as $productId) {
+            $this->testHelper->addProduct($productId, rand(1,3));
+        }
+
+        $quote->setTotalsCollectedFlag(false);
+        $quote->collectTotals()->save();
+
+        $reflectedShippingController = new ReflectionClass($this->_shippingController);
```

Comment:
how about we simply make `getEstimateCacheIdentifier` public? it seems testing get a bit complicated due to boilerplate code for reflection

Reply:
The argument is fair.  In general, I wouldn't want the testing framework dictating the implementation of classes that it is testing, but in this case, there is no significant reason why `getEstimateCachIdentifier` shouldn't be public.